packages/core/database/lib/entity-manager/index.js

```diff
@@ -458,25 +495,28 @@ const createEntityManager = (db) => {
           await this.createQueryBuilder(target)
             .update({ [attribute.joinColumn.referencedColumn]: id })
             // NOTE: works if it is an array or a single id
-            .where({ id: data[attributeName] })
+            .where({ id: relIdsToAdd })
             .execute();
         }
 
         if (attribute.joinTable) {
           // need to set the column on the target
 
           const { joinTable } = attribute;
-          const { joinColumn, inverseJoinColumn } = joinTable;
-
-          if (isOneToAny(attribute) && isBidirectional(attribute)) {
-            await this.createQueryBuilder(joinTable.name)
-              .delete()
-              .where({ [inverseJoinColumn.name]: castArray(data[attributeName]) })
-              .where(joinTable.on || {})
-              .execute();
+          const { joinColumn, inverseJoinColumn, orderColumnName, inverseOrderColumnName } =
+            joinTable;
+          const select = [joinColumn.name];
+          if (isAnyToMany(attribute)) {
+            select.push(orderColumnName);
           }
```

Comment:
I hate to be this type of person. But I am missing some comments here giving context 😬
I understand you are selecting the order column only if the relationship is between many others? So then you increment that order.